operators/src/pro/cache/tile_cache.rs

```diff
@@ -29,16 +30,425 @@ pub struct TileCacheBackend {
+    fn remove_bytes(&mut self, bytes: usize) {
+        debug_assert!(
+            self.byte_size_used >= bytes,
+            "removing more bytes {} than used {}",
+            bytes,
+            self.byte_size_used
+        );
+        self.byte_size_used -= bytes;
+    }
+
+    #[inline]
+    fn remove_element_bytes<T: ByteSize>(&mut self, element: &T) {
+        self.remove_bytes(element.byte_size());
+    }
+
+    #[inline]
+    pub fn try_add_bytes(&mut self, bytes: usize) -> Result<(), CacheError> {
+        ensure!(
+            self.can_fit_bytes(bytes),
+            crate::pro::cache::error::NotEnoughSpaceInCache
+        );
+        self.add_bytes(bytes);
+        Ok(())
+    }
+
+    #[inline]
+    pub fn try_add_element_bytes<T: ByteSize>(&mut self, element: &T) -> Result<(), CacheError> {
+        let bytes = element.byte_size();
+        self.try_add_bytes(bytes)
+    }
+
+    #[inline]
+    pub fn try_remove_bytes(&mut self, bytes: usize) -> Result<(), CacheError> {
+        ensure!(
+            self.byte_size_used >= bytes,
+            crate::pro::cache::error::NegativeSizeOfCache
+        );
+        self.remove_bytes(bytes);
+        Ok(())
+    }
+
+    #[inline]
+    pub fn try_remove_element_bytes<T: ByteSize>(&mut self, element: &T) -> Result<(), CacheError> {
+        let bytes = element.byte_size();
+        self.try_remove_bytes(bytes)
+    }
+
+    #[inline]
+    fn add_bytes_allow_overflow(&mut self, bytes: usize) {
+        self.byte_size_used += bytes;
+        if self.byte_size_used > self.byte_size_total {
+            trace!(
+                "overflowing cache size by {} bytes, total size: {}, added bytes: {}",
+                self.byte_size_used - self.byte_size_total,
+                self.byte_size_total,
+                bytes
+            );
+        }
+    }
+}
+
+type CacheSizeLruRef<'a> = (
+    &'a mut HashMap<CacheEntryId, CacheEntry>,
+    &'a mut TileCacheBackendSize,
+    &'a mut LruCache<CacheEntryId, CanonicOperatorName>,
+);
+
+type LandingZoneSizeRef<'a> = (
+    &'a mut HashMap<QueryId, LandingZoneEntry>,
+    &'a mut TileCacheBackendSize,
+);
+
+impl TileCacheBackend {
+    fn cache_and_size_and_lru_mut(&mut self, key: &CanonicOperatorName) -> Option<CacheSizeLruRef> {
+        self.operator_caches
+            .get_mut(key)
+            .map(|cache| (&mut cache.entries, &mut self.cache_size, &mut self.lru))
+    }
+
+    fn landing_zone_and_size_mut(
+        &mut self,
+        key: &CanonicOperatorName,
+    ) -> Option<LandingZoneSizeRef> {
+        self.operator_caches
+            .get_mut(key)
+            .map(|cache| (&mut cache.landing_zone, &mut self.landing_zone_size))
+    }
+
+    pub fn query_cache<T>(
+        &mut self,
+        key: &CanonicOperatorName,
+        query: &RasterQueryRectangle,
+    ) -> Option<CacheTileStream<T>>
+    where
+        T: Pixel + Cachable,
+    {
+        let (cache, _cache_size, cache_lru) = self.cache_and_size_and_lru_mut(key)?;
```

Comment:
make a custom method withotu _cache_size if it's ignored?

Reply:
i removed the methods

Reply:
[0739e8b](https://github.com/geo-engine/geoengine/pull/819/commits/0739e8b3e781ca3c6444c168e43f815b21ff6d9d)